File: dart/luzid-sdk/lib/src/api/meta.dart
import 'package:luzid_grpc/luzid_grpc.dart';
import 'package:luzid_grpc_client/luzid_grpc_client.dart';
import 'package:luzid_sdk/src/core/utils.dart';

export 'package:luzid_grpc/luzid_grpc.dart' show Meta;

class LuzidMeta {
  final MetaClient _client;

  LuzidMeta(LuzidGrpcClient client) : _client = client.meta;

  /// Performs a meta request to get information about the Luzid instance.
  Future<Meta> getMeta() async {
    final res = await _client.getMeta();
    return unwrap(res, 'Luzid meta.getMeta').meta;
  }
}

File: ts/packages/luzid-sdk/src/api/meta.ts
import { LuzidGrpcClient } from '@luzid/grpc-client'
import { Successful, unwrap } from '../core/utils'
import { MetaResponse } from '@luzid/grpc'

export { MetaResponse } from '@luzid/grpc'

export class LuzidMeta {
  constructor(private readonly client: LuzidGrpcClient) {}

  /**
   * Get the meta information of the Luzid backend.
   */
  async getMeta(): Promise<Successful<MetaResponse>> {
    const res = await this.client.meta.getMeta()
    return unwrap(res, 'Luzid meta.getMeta')
  }
}
